pkg/daemon/ceph/client/subvolumegroup.go

```diff
@@ -17,32 +17,103 @@ limitations under the License.
 package client
 
 import (
+	"encoding/json"
 	"fmt"
 	"strconv"
+	"syscall"
 
 	"github.com/pkg/errors"
 	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
 	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/util/exec"
 	"k8s.io/apimachinery/pkg/types"
 )
 
 // CreateCephFSSubVolumeGroup create a CephFS subvolume group.
 // volName is the name of the Ceph FS volume, the same as the CephFilesystem CR name.
-func CreateCephFSSubVolumeGroup(context *clusterd.Context, clusterInfo *ClusterInfo, volName, groupName string) error {
+func CreateCephFSSubVolumeGroup(context *clusterd.Context, clusterInfo *ClusterInfo, volName, groupName string, svgSpec *cephv1.CephFilesystemSubVolumeGroupSpec) error {
 	logger.Infof("creating cephfs %q subvolume group %q", volName, groupName)
-	//  [--pool_layout <data_pool_name>] [--uid <uid>] [--gid <gid>] [--mode <octal_mode>]
+	// [<size:int>] [--pool_layout <data_pool_name>] [--uid <uid>] [--gid <gid>] [--mode <octal_mode>]
 	args := []string{"fs", "subvolumegroup", "create", volName, groupName}
+	if svgSpec != nil {
+		if svgSpec.Quota != nil {
+			// convert the size to bytes as ceph expect the size in bytes
+			args = append(args, fmt.Sprintf("--size=%d", svgSpec.Quota.Value()))
+		}
+		if svgSpec.DataPoolName != "" {
+			args = append(args, fmt.Sprintf("--pool_layout=%s", svgSpec.DataPoolName))
+		}
+	}
+
+	_, err := getCephFSSubVolumeGroupInfo(context, clusterInfo, volName, groupName)
+	if err != nil {
+		// return error other than not found.
+		if code, ok := exec.ExitStatus(err); ok && code != int(syscall.ENOENT) {
+			return errors.Wrapf(err, "failed to create subvolume group %q in filesystem %q", groupName, volName)
+		}
+	}
+
+	// if the subvolumegroup exists, resize the subvolumegroup
+	if err == nil && svgSpec != nil && svgSpec.Quota != nil {
+		err = resizeCephFSSubVolumeGroup(context, clusterInfo, volName, groupName, svgSpec)
```

Comment:
Can we skip calling resize if the size didn't change?

Reply:
Yes we can do but this is no op if the size doesn't change. let me know if you want me to add that check

Reply:
Since we called `getCephFSSubVolumeGroupInfo()`, we can easily compare if there was a change, right? It's nice to skip API calls if it's simple to compare.